js/state/siteUtil.js

```diff
@@ -417,6 +417,10 @@ module.exports.removeSite = function (state, siteDetail, tag, reorder = true, sy
     site = site.set('tags', tags)
     return state.setIn(stateKey, site)
   } else {
+    if (!tag && module.exports.isHistoryEntry(siteDetail)) {
+      // Delete the site from history
+      return state.deleteIn(stateKey)
```

Comment:
We don't delete history for a reason which is to prevent https://github.com/brave/browser-laptop/issues/7238 from happening. This line will cause the regression of #7238.
Maybe we should revisit this after #10136 landed